app/ami.py

```diff
@@ -209,10 +209,11 @@ def estimate_monthly_ami_price(ami_type: str, block_device_mappings: list, ami_n
 
 
 def is_ami_registered(ami_id: str, ) -> bool:
-    # Use global variable to make API call once instead of every time the method is called since call is costly
-    global ORG_AMIS
-    if not ORG_AMIS:
-        ec2_client = boto3.client('ec2')
-        ORG_AMIS = ec2_client.describe_images(Owners=['self'])
-    # Iterate through each image owned by organization to check if ami id exists, if it does it is registered
-    return True if [i['ImageId'] for i in ORG_AMIS['Images'] if i['ImageId'] == ami_id] else False
+    ec2 = boto3.resource('ec2')
+    is_registered = True
+    # Retrieve name of AMI, if ClientError or AttributeError is thrown, the AMI does not exist
+    try:
+        ec2.Image(ami_id).name
+    except (botocore.exceptions.ClientError, AttributeError):
```

Comment:
`AttributeError` seems suspicious...

Reply:
I agree - I don't know if you saw my previous comment, but I discovered a bug with an open ticket [here](https://github.com/boto/boto3/issues/925) on the boto3 project’s GitHub page describing how None type is returned for deregistered amis, and the error that is thrown is: `AttributeError: 'NoneType' object has no attribute 'get'`.
If its too suspicious I can always go back to the other method of getting a list of our AMIs and checking them, what do you think? I'm alright with using the `AttributeError` with the `ClientError` because I can't anticipate serious ramifications, but I may be overlooking something.